src/Type/ObjectType.php

```diff
@@ -1320,4 +1320,13 @@ public function tryRemove(Type $typeToRemove): ?Type
 		return null;
 	}
 
+	public function exponentiate(Type $exponent): Type
+	{
+		$object = new ObjectWithoutClassType();
+		if (!$exponent instanceof NeverType && !$object->isSuperTypeOf($this)->no() && !$object->isSuperTypeOf($exponent)->no()) {
```

Comment:
Objects can be exponentiated? Huh?

Reply:
yep, see https://github.com/phpstan/phpstan-src/blob/285b49f2072e18ffe9ff72f221151512b8a92fe6/tests/PHPStan/Analyser/data/pow.php#L17-L25